packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -62,11 +62,11 @@ abstract class Listenable {
   /// Return a [Listenable] that triggers when any of the given [Listenable]s
   /// themselves trigger.
   ///
-  /// The list must not be changed after this method has been called. Doing so
-  /// will lead to memory leaks or exceptions.
+  /// [listenables] must not be changed after this method has been called.
+  /// Doing so will lead to memory leaks or exceptions.
   ///
-  /// The list may contain nulls; they are ignored.
-  factory Listenable.merge(List<Listenable?> listenables) = _MergingListenable;
+  /// The iterable may contain nulls; they are ignored.
+  factory Listenable.merge(Iterable<Listenable?> listenables) = _MergingListenable;
```

Comment:
Can you add a simple test for this change (e.g. a test that passes in a statically typed Iterable and would break if we ever revert this change)?

Reply:
Sounds good! There's already a test that uses multiple `List`s, so I changed one of them to a `Set`.